File: Assets/Scripts/Constant/ErrorClass.cs
using System.Collections.Generic;

namespace Rafting
{
    /// <summary>
    /// 서버와 클라이언트 간에 공유되는 에러 코드 및 메시지를 정의하는 클래스입니다.
    /// Java의 ErrorClass를 C#으로 변환한 것입니다.
    /// </summary>
    public static class ErrorClass
    {
        public const string ALLOK = "E000";
        public const string GAME_SERVER_FULL = "E700";
        public const string ROOM_NOT_AVAILABLE = "E701";
        public const string ROOM_ALREADY_FULL = "E702";
        public const string ROOM_ALREADY_EXIST = "E703";
        public const string DATA_EMPTY = "E797";
        public const string DATA_MISMATCH = "E798";
        public const string SERVER_UNKNOWN_ERROR = "E799";

        /// <summary>
        /// 에러 코드에 해당하는 다국어 메시지를 반환합니다.
        /// </summary>
        /// <param name="errorCode">에러 코드</param>
        /// <returns>0: 영어 메시지, 1: 한국어 메시지</returns>
        public static Dictionary<string, string> GetErrorMessage(string errorCode)
        {
            var messages = new Dictionary<string, string>();

            switch (errorCode)
            {
                case ALLOK:
                    messages["en"] = "OK";
                    messages["ko"] = "-";
                    break;
                case GAME_SERVER_FULL:
                    messages["en"] = "Unknown Error has been occured.";
                    messages["ko"] = "-";
                    break;
                case ROOM_NOT_AVAILABLE:
                    messages["en"] = "The room is not available. Please try to enter another room.";
                    messages["ko"] = "-";
                    break;
                case ROOM_ALREADY_FULL:
                    messages["en"] = "The room is already full. Please try to enter another room.";
                    messages["ko"] = "-";
                    break;
                case ROOM_ALREADY_EXIST:
                    messages["en"] = "The room already exist.";
                    messages["ko"] = "-";
                    break;
                case DATA_EMPTY:
                    messages["en"] = "Data search result is empty.";
                    messages["ko"] = "-";
                    break;
                case DATA_MISMATCH:
                    messages["en"] = "Some wrong data have been found. You need to restart the game.";
                    messages["ko"] = "-";
                    break;
                case SERVER_UNKNOWN_ERROR:
                    messages["en"] = "Unknown Error has been occured.";
                    messages["ko"] = "-";
                    break;
                default:
                    messages["en"] = "-";
                    messages["ko"] = "-";
                    break;
            }
            return messages;
        }
    }
}

File: TempServerSource/src/weaving/playasone/ErrorClass.java
package weaving.playasone;

import java.util.ArrayList;

public class ErrorClass {
	public static final String ALLOK 				= "E000";
	
	public static final String GAME_SERVER_FULL 	= "E700";
	public static final String ROOM_NOT_AVAILABLE 	= "E701";
	public static final String ROOM_ALREADY_FULL 	= "E702";
	public static final String ROOM_ALREADY_EXIST 	= "E703";

	public static final String DATA_EMPTY 			= "E797";
	public static final String DATA_MISMATCH 		= "E798";
	public static final String SERVER_UNKNOWN_ERROR = "E799";
	
	/*
	 * Error Message in Array
	 * Index 0 -> English
	 * Index 1 -> Korean
	 */
	public static ArrayList<String> GetErrorMessage(String ErrorCode)
	{
		ArrayList<String> temp = new ArrayList<String>();
		
		switch (ErrorCode)
		{
			case "E000":
				temp.add("OK");
				temp.add("-");
				break;
			case "E700":
				temp.add("Unknown Error has been occured.");
				temp.add("-");
				break;
			case "E701":
				temp.add("The room is not available. Please try to enter another room.");
				temp.add("-");
				break;
			case "E702":
				temp.add("The room is already full. Please try to enter another room.");
				temp.add("-");
				break;
			case "E703":
				temp.add("The room already exist.");
				temp.add("-");
				break;
			case "E797":
				temp.add("Data search result is empty.");
				temp.add("-");
				break;
			case "E798":
				temp.add("Some wrong data have been found. You need to restart the game.");
				temp.add("-");
				break;
			case "E799":
				temp.add("Unknown Error has been occured.");
				temp.add("-");
				break;
			default:
				temp.add("-");
				temp.add("-");
				break;
		}
		
		return temp;
	}
}
